py3status/modules/usbguard.py

```diff
@@ -0,0 +1,350 @@
+                params[key] = value
+            devices_array.append(params)
+
+        for index, device in sorted(enumerate(devices_array), reverse=False):
+            response[index + 1] = device
+
+        # self.py3.log('# response')
+        # self.py3.log(response)
+        
+        return response
+
+    def _toggle_permanant(self):
+        self.is_permanent = not self.is_permanent
+
+    def post_config_hook(self):
+        self._init_dbus()
+        self.all_devices = {}
+        self.data = {}
+        self.usbguard_data = {}
+        self.is_permanent = False
+
+        # init placeholders
+        available_placeholders = [
+            'id',
+            'name',
+            'via_port',
+            'hash',
+            'parent_hash',
+            'serial',
+            'with_interface',
+            'format_all_devices',
+        ]
+        self.placeholders = {}
+        placeholders = self.py3.get_placeholders_list(
+            self.format_notification + self.format_all_devices
+        )
+        for placeholder in available_placeholders:
+            if placeholder in placeholders:
+                self.placeholders[placeholder] = None
+        self.placeholders['usbguard_id'] = None
+
+        self.killed = threading.Event()
+        UsbguardListener(self).start()
+
+    def _set_policy(self, action, index):
+        if action and index and index != 'sep':
+    
+            targets = {'allow': 0, 'block': 1, 'reject': 2}
+
+            self.py3.log('# data')
+            self.py3.log(self.data)
+            self.py3.log('# index')
+            self.py3.log(index)
+            self.py3.log('# action')
+            self.py3.log(action)
+            self.py3.log('# usbguard data')
+            self.py3.log(self.usbguard_data)
+            self.py3.log('# all devices')
+            self.py3.log(self.all_devices)
+            
+            # notifications
+            if self.format_notification:
+                format_notification = self.data[index]
+                format_notification['action'] = action
+                format_notification = self.py3.safe_format(
+                    self.format_notification, format_notification
+                )
+                notification = self.py3.get_composite_string(format_notification)
+                self.py3.notify_user(
+                    notification,
+                    title='USBGuard',
+                    icon='/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg',
+                )
+
+            # apply policy
+            if 'usbguard_id' in self.data and index != 'sep':
+                self.proxy.applyDevicePolicy(
+                    self.data['usbguard_id'], targets[action], self.is_permanent
+                )
+
+    def _manipulate_all_devices(self, data):
+        format_all_devices = []
+        format_device_separator = self.py3.safe_format(self.format_device_separator)
+        self.py3.composite_update(format_device_separator, {'index': 'sep'})
```

Comment:
This index line from `google_calendar` is not ideal code. Remove it. We add `index` to the ones we care about. We don't care about getting click events of users clicking on separators.